### PR TITLE
Sending tx on Android(without approval yet)

### PR DESCRIPTION
**Pull request by SergeyZhukovsky**
Resolves https://github.com/brave/brave-browser/issues/18072

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java [hunk 1]
```diff
@@ -247,6 +300,56 @@ public void onClick(View v) {
                 }
             });
         }
+
+        btnBuySendSwap.setOnClickListener(v -> {
+            if (mActivityType == ActivityType.SEND) {
+                // TODO(sergz): token transfers via a token contract
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java:306]: https://github.com/brave/brave-browser/issues/18073

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java [hunk 2]
```diff
@@ -247,6 +300,56 @@ public void onClick(View v) {
...
+                String to = toValueText.getText().toString();
+                if (to.isEmpty()) {
+                    // TODO(sergz): some address validation
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java:309]: https://github.com/brave/brave-browser/issues/18074

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java [hunk 3]
```diff
@@ -247,6 +300,56 @@ public void onClick(View v) {
...
+                    return;
+                }
+                TextView fromValueText = findViewById(R.id.from_value_text);
+                String value = fromValueText.getText().toString();
+                // TODO(sergz): Some kind of validation that we have enough balance
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java:314]: https://github.com/brave/brave-browser/issues/18074

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java [hunk 4]
```diff
@@ -247,6 +300,56 @@ public void onClick(View v) {
...
+                // TODO(sergz): Get gas price
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java:315]: https://github.com/brave/brave-browser/issues/17978
https://github.com/brave/brave-browser/issues/18075

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java [hunk 5]
```diff
@@ -247,6 +300,56 @@ public void onClick(View v) {
...
+                TxData data = Utils.getTxData("0x1", "0x2FBF9BEDA5F0", "0x2FBF9BEDA5F1", to,
+                        Utils.toHexWei(value), new byte[0]);
+                Spinner accountSpinner = findViewById(R.id.accounts_spinner);
+                String from = mCustomAccountAdapter.getTitleAtPosition(
+                        accountSpinner.getSelectedItemPosition());
+                if (mEthTxController != null) {
+                    mEthTxController.addUnapprovedTransaction(data, from,
+                            (success, tx_meta_id)
+                                    -> {
+                                            // TODO(sergz): Add an observer and approve transaction
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java:325]: https://github.com/brave/brave-browser/issues/18076

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java [hunk 6]
```diff
@@ -132,24 +138,70 @@ protected void triggerLayoutInflation() {
...
+                return;
+            }
+            TextView fromBalanceText = findViewById(R.id.from_balance_text);
+            mConvertedBalance = Utils.fromHexWei(balance);
+            fromBalanceText.setText(getText(R.string.crypto_wallet_balance) + " "
+                    + String.format("%.4f", mConvertedBalance));
+        });
+    }
+
     private int getIndexOf(Spinner spinner, String chain_id) {
```
Comment (deeppandya) [inline android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java:205]: nit `chain_id` should be chainId. just the naming convention

**Review by deeppandya — APPROVED** [review]
LGTM

**Review by alexsafe — APPROVED** [review]

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java (hunk 5 above)
Comment (bbondy) [reply android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java:325]: btw here I don't think you'll do anything. I think you want the observer to be independent of this and respond by opening the UI to approve. When the UI action for approval happens you can call to approve the transaction.

**Review by bbondy — APPROVED** [review]

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java (hunk 5 above)
Comment (SergeyZhukovsky) [reply android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java:325]: yep, just posted it somewhere to not forget

**Pull request merged**